Gems/ROS2/Code/Source/Sensor/SensorConfiguration.h

```diff
@@ -0,0 +1,36 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
+
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    /// General configuration for sensors
+    struct SensorConfiguration
+    {
+    public:
+        AZ_RTTI(SensorConfiguration, "{4755363D-0B5A-42D7-BBEF-152D87BA10D7}");
+
+        SensorConfiguration() = default;
+        virtual ~SensorConfiguration() = default;
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        // TODO - publishing-related data
+        AZStd::string m_topic; // TODO - apply namespace, default to standard names per type, validation
```

Comment:
The single topic per sensor is a strong assumption. It can have more than one (like mentioned [here](https://github.com/RobotecAI/o3de/pull/8/files#r850548847) by PRy), but I think it can also have none, for example when using services (imagine a sensor which is being pulled rather than publishing data).

Reply:
It is good for now. I responded here: https://github.com/RobotecAI/o3de/pull/8#discussion_r850629432